server/src/main/java/org/opensearch/tasks/TaskCompleteResourceInfo.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TaskCompleteResourceInfo {
+    private final Map<TaskStatsType, TaskResourceInfo> resourceInfo = new HashMap<>();
+    private boolean isActive;
+
+    public TaskCompleteResourceInfo(boolean isActive, TaskStatsType statsType, TaskResourceMetric... taskResourceMetrics) {
+        this.isActive = isActive;
+        this.resourceInfo.put(statsType, new TaskResourceInfo(taskResourceMetrics));
+    }
+
+    public void update(boolean isActive, TaskStatsType statsType, TaskResourceMetric... taskResourceMetrics) {
```

Comment:
Whats the logic of moving the active to false. Is this required by the last worker thread to be changing the active state to inactive. If yes doesn't look like thread safe to me.

Reply:
Use Case:

- When a runnable starts it'll send an update that thread has started to work on this task so these are the starting values for resource utilisation.
- When runnable is about to finish it sends another update. 

Both these updates will be sent by same worker thread. 

We need to make some changes to make all the state updates thread safe though